vivarium/core/emitter.py

```diff
@@ -425,6 +439,9 @@ def get_history_data_db(
     for document in cursor:
         assert document.get('assembly_id'), \
             "all database documents require an assembly_id"
+        if f and query:
+            for field in query:
+                document["data"] = apply_func(document["data"], field, f)
```

Comment:
I can imagine that in lots of cases, users might want to only apply the function to some of the fields they want to retrieve. Can we support that? Some ideas:

1. Instead of just providing a function `f`, provide a dictionary mapping paths to functions.
2. Pass the path `field` to the function `f` so the function can behave differently depending on what field it's operating on

Reply:
I added a `func_dict` arg that does this.